tests/countries/test_sierra_leone.py

```diff
@@ -0,0 +1,256 @@
+            "2021-07-20",
+            "2022-07-09",
+            "2023-06-28",
+            "2024-06-16",
+        )
+        self.assertHolidayName(name, self.no_estimated_holidays, range(1962, 2050))
+
+    def test_2024(self):
+        self.assertHolidays(
+            SierraLeone(years=2024),
+            ("2024-01-01", "New Year's Day"),
+            ("2024-02-18", "Armed Forces Day"),
+            ("2024-02-19", "Armed Forces Day (observed)"),
+            ("2024-03-08", "International Women's Day"),
+            ("2024-03-29", "Good Friday"),
+            ("2024-04-01", "Easter Monday"),
+            ("2024-04-10", "Eid al-Fitr"),
+            ("2024-04-27", "Independence Day"),
+            ("2024-04-29", "Independence Day (observed)"),
+            ("2024-05-01", "International Worker's Day"),
+            ("2024-06-16", "Eid al-Adha"),
+            ("2024-06-17", "Eid al-Adha (observed)"),
+            ("2024-09-15", "Prophet's Birthday"),
+            ("2024-09-16", "Prophet's Birthday (observed)"),
+            ("2024-12-25", "Christmas Day"),
+            ("2024-12-26", "Boxing Day"),
+        )
+
+    def test_l10n_default(self):
+        self.assertLocalizedHolidays(
+            ("2025-01-01", "New Year's Day"),
+            ("2025-02-18", "Armed Forces Day"),
+            ("2025-03-08", "International Women's Day"),
+            ("2025-03-10", "International Women's Day (observed)"),
+            ("2025-03-30", "Eid al-Fitr (estimated)"),
+            ("2025-03-31", "Eid al-Fitr (observed, estimated)"),
+            ("2025-04-18", "Good Friday"),
+            ("2025-04-21", "Easter Monday"),
+            ("2025-04-27", "Independence Day"),
+            ("2025-04-28", "Independence Day (observed)"),
+            ("2025-05-01", "International Worker's Day"),
+            ("2025-06-06", "Eid al-Adha (estimated)"),
+            ("2025-09-04", "Prophet's Birthday (estimated)"),
+            ("2025-12-25", "Christmas Day"),
+            ("2025-12-26", "Boxing Day"),
+        )
+
+    def test_l10n_en_us(self):
+        self.assertLocalizedHolidays(
+            "en_US",
+            ("2025-01-01", "New Year's Day"),
+            ("2025-02-18", "Armed Forces Day"),
+            ("2025-03-08", "International Women's Day"),
+            ("2025-03-10", "International Women's Day (observed)"),
+            ("2025-03-30", "Eid al-Fitr (estimated)"),
+            ("2025-03-31", "Eid al-Fitr (observed, estimated)"),
+            ("2025-04-18", "Good Friday"),
+            ("2025-04-21", "Easter Monday"),
+            ("2025-04-27", "Independence Day"),
+            ("2025-04-28", "Independence Day (observed)"),
+            ("2025-05-01", "Labor Day"),
+            ("2025-06-06", "Eid al-Adha (estimated)"),
+            ("2025-09-04", "Prophet's Birthday (estimated)"),
+            ("2025-12-25", "Christmas Day"),
+            ("2025-12-26", "Boxing Day"),
+        )
+
+    def test_new_years_day_observed(self):
+        dt = (
```

Comment:
> Also need to add check for "observed" for all holidays where it can be (as for Independence Day).

> as for Independence Day

You have one properly done method. Why not make other methods like that?